47758 adds new aws add-on topic (#784)
replicatedhq/kurl.sh: src/markdown-pages/add-ons/aws.md

---
path: "/docs/add-ons/aws"
date: "2022-05-10"
linktitle: "AWS Add-On"
weight: 31
title: "AWS Add-On"
addOn: "aws"
---

> The AWS add-on allows you to customize the configuration of the [Kubernetes (kubeadm)](/docs/addon-ons/kubernetes) control plane to interface to the AWS cloud provider service for storage classes. This means that the apiserver, controllermanager, and kubelet will all be started with the extra argument specifying that the cloud provider is AWS. This enables you to have Kubernetes provision [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)directly from the default storage classes available from AWS to the machine that kURL is installed on running in AWS. If needed, the advanced option `ExcludeStorageClass` may be used to exclude the `aws-ebs` provisioner storage class provided by the AWS add-on. For more information about the AWS cloud provider for Kubernetes, see the [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws) repository in GitHub.

Reviewer: ```suggestion
The AWS add-on allows you to customize the configuration of the [Kubernetes (kubeadm)](/docs/addon-ons/kubernetes) control plane to interface to the AWS cloud provider service for storage classes. This means that the apiserver, controllermanager, and kubelet will all be started with the extra argument specifying that the cloud provider is AWS. This enables you to have Kubernetes provision [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) directly from the default storage classes available from AWS to the machine that kURL is installed on running in AWS. If needed, the advanced option `ExcludeStorageClass` may be used to exclude the `aws-ebs` provisioner storage class provided by the AWS add-on. For more information about the AWS cloud provider for Kubernetes, see the [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws) repository in GitHub.
```

> The AWS add-on allows you to customize the configuration of the [Kubernetes (kubeadm)](/docs/addon-ons/kubernetes) control plane to interface to the AWS cloud provider service for storage classes. This means that the apiserver, controllermanager, and kubelet will all be started with the extra argument specifying that the cloud provider is AWS. This enables you to have Kubernetes provision [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) directly from the default storage classes available from AWS to the machine that kURL is installed on running in AWS. If needed, the advanced option `ExcludeStorageClass` may be used to exclude the `aws-ebs` provisioner storage class provided by the AWS add-on. For more information about the AWS cloud provider for Kubernetes, see the [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws) repository in GitHub.

Reviewer 2: this is too storage class specific. enabling the aws addon adds more features than just storage classes

Reviewer 2: something more like this. lifted straight from the docs

> The AWS cloud provider provides the interface between a Kubernetes cluster and AWS service APIs. This project allows a Kubernetes cluster to provision, monitor and remove AWS resources necessary for operation of the cluster.

> The AWS add-on allows you to customize the configuration of the [Kubernetes (kubeadm)](/docs/addon-ons/kubernetes) control plane to interface to the AWS cloud provider service for storage classes. This means that the apiserver, controllermanager, and kubelet will all be started with the extra argument specifying that the cloud provider is AWS. This enables you to have Kubernetes provision [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) directly from the default storage classes available from AWS to the machine that kURL is installed on running in AWS. If needed, the advanced option `ExcludeStorageClass` may be used to exclude the `aws-ebs` provisioner storage class provided by the AWS add-on. For more information about the AWS cloud provider for Kubernetes, see the [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws) repository in GitHub.

Reviewer 2: ```suggestion
The AWS add-on enables the [AWS cloud provider](https://github.com/kubernetes/cloud-provider-aws) integration with the [Kubernetes (kubeadm) add-on](/docs/addon-ons/kubernetes). The AWS cloud provider provides the interface between a Kubernetes cluster and AWS service APIs. This project enables dynamic provisioning of Elastic Block Store (EBS) volumes as well as dynamic provisioning/configuration of Elastic Load Balancers (ELBs) for exposing Kubernetes Service objects.

Because the AWS cloud controller manager performs some tasks on behalf of the operator, like creating an ELB or an EBS volume, you will need to create IAM policies and assign them to your EC2 instances. Please see the [prerequisites guide](https://kubernetes.github.io/cloud-provider-aws/prerequisites/) for the permissions needed.
```

## Advanced Install Options

```yaml
spec:
  aws:
    version: 0.1.0
    excludeStorageClass: false
```

flags-table

## Requirements and Limitations

> * When used, a volume provisioner add-on does not need to be included in the kURL specification as [AWS EBS](https://aws.amazon.com/ebs/) will be used.

Reviewer 2: ```suggestion
* When enabled, a volume provisioner add-on is not necessary in the kURL specification as the [AWS EBS](https://aws.amazon.com/ebs/) volume provisioner may be used.
```

> * This add-on currently only configures the above mentioned portions of the Kubernetes control plane and only uses AWS for storage provisioner services.

Reviewer 2: remove this

Author: To confirm: remove line 26? (This add-on currently only configures the above mentioned portions of the Kubernetes control plane and only uses AWS for storage provisioner services.)

Always hard to tell with Github comments

> * This add-on is only intended to be used on AWS.  

Reviewer 2: additionally i would add

```
* This add-on currently only works with the [Kubernetes (kubeadm) add-on](/docs/addon-ons/kubernetes). Support has not yet been added for [K3s](/docs/addon-ons/k3s) or [RKE2](/docs/addon-ons/rke2).
```

---
path: "/docs/add-ons/aws"
date: "2022-05-10"
linktitle: "AWS Add-On"
weight: 31
title: "AWS Add-On"
addOn: "aws"
---

The AWS add-on enables the Amazon Web Services (AWS) cloud provider integration with the Kubernetes (kubeadm) kURL add-on. For information about the kubeadm add-on, see [Kubernetes (kubeadm) Add-On](/docs/addon-ons/kubernetes).

The AWS cloud provider provides the interface between a Kubernetes cluster and AWS service APIs. It enables dynamic provisioning of Elastic Block Store (EBS) volumes as well as dynamic provisioning and configuration of Elastic Load Balancers (ELBs) for exposing Kubernetes Service objects.

> For more information about the AWS cloud provider, see the [AWS cloud provider](https://github.com/kubernetes/cloud-provider-aws) repository in GitHub.

Reviewer 2: ```suggestion
For more information about the AWS cloud provider, see the [AWS cloud provider](https://cloud-provider-aws.sigs.k8s.io/) online documentation.
```

## Requirements and Limitations

> * When used, a volume provisioner add-on does not need to be included in the kURL specification as [AWS EBS](https://aws.amazon.com/ebs/) will be used.

Reviewer: I wouldn't call this a limitation

Reviewer 3: Agree, though felt like it was more or less a requirement or something that should be mentioned. Maybe requirements and limitations isn't the best place. @paigecalvert wdyt?

Author: Oh good call--I created a separate section for this info with my latest commit. See whatcha think https://github.com/replicatedhq/kurl.sh/pull/784/commits/1aa2043e61be3cc190ae0dac5703df6626b09da5
